spatialdata/_core/elements.py

```diff
@@ -149,26 +192,68 @@ def from_path(
 
     @property
     def shape(self) -> Tuple[int, ...]:
-        return self.data.shape  # type: ignore[no-any-return]
+        return self.data.obsm["spatial"].shape  # type: ignore[no-any-return]
 
 
 class Polygons(BaseElement):
-    def __init__(self, polygons: Any, transform: Transform) -> None:
+    def __init__(self, polygons: Any, transform: BaseTransformation) -> None:
         self.data: Any = polygons
         self.transforms = transform
         super().__init__()
 
     @staticmethod
     def parse_polygons(data: AnnData, transform: Optional[Any] = None) -> "Polygons":
-        data, transform = parse_dataset(data, transform)
+        # data, transform = parse_dataset(data, transform)
+        if transform is None:
+            transform = Identity()
         return Polygons(data, transform)
 
+    @staticmethod
+    def tensor_to_string(x: ArrayLike) -> str:
+        s = repr(x).replace("\n", "").replace(" ", "")[len("array(") : -1]
+        # consistently check
+        y = eval(s)
+        assert np.allclose(x, y)
+        return s
+
+    @staticmethod
+    def string_to_tensor(s: str) -> Union[ArrayLike, None]:
+        pattern = r"^\[(?:\[[0-9]+\.[0-9]+,[0-9]+\.[0-9]+\],?)+\]$"
+        if re.fullmatch(pattern, s) is not None:
+            x: ArrayLike = np.array(eval(s))
+            return x
+
+    @staticmethod
+    def anndata_from_geojson(path: str) -> AnnData:
+        with open(path) as f:
+            j = json.load(f)
+
+        names = []
+        coordinates = []
+        assert "geometries" in j
+        for region in j["geometries"]:
+            if region["type"] == "Polygon":
+                names.append(region["name"])
+                vertices: ArrayLike = np.array(region["coordinates"])
+                vertices = np.squeeze(vertices, 0)
+                assert len(vertices.shape) == 2
+                coordinates.append(vertices)
+            else:
+                print(f'ignoring "{region["type"]}" from geojson')
+
+        string_coordinates = [Polygons.tensor_to_string(c) for c in coordinates]
+        a = AnnData(shape=(len(names), 0), obs=pd.DataFrame({"name": names, "spatial": string_coordinates}))
+        return a
+
```

Comment:
approving this but ofc will need rewrite

Reply:
Agree, it needs to be rewritten. Also, I moved it spatialdata-io (keeping it here in this pr).